### ISSUE TITLE
Implement testing logic to compare the Shoot-spec generated by provisioner with the shoot-spec generated by KIM

### ISSUE THREAD

**Issue by tobiscr** [kyma-project/infrastructure-manager#185]
**Reason**
We want to ensure that our migration from Provisioner to Kyma-Infrastructure-Manager will go without any issues. To create a shoot cluster in Gardener, we have to fill [gardener's shoot spec](https://github.com/gardener/gardener/blob/master/example/90-shoot.yaml). 

**How**
Ideas on how to achieve that:

### 1. Comparing shoot spec of created clusters
1. Prepare two KEB deployments. First should be still configured to use the old provisioner, and the second should use KIM.
2. Send the very same instance creation requests to KEB.
3. Compare the shoot specs of created clusters.

### 2. Comparing created shoot spec, before sending them to Gardener
1. Create a feature flag in the provisioner for dumping shoot spec before sending it to gardener
2. Save those specs 
3. Compare shoot specs generated in KIM with those generated in provisioner

**Acceptance criteria**
- [x] Each KEB's plan should be covered 
- [x] compare (manually) the Shoot-spec generated by provisioner with the shoot-spec generated by KIM
- [x] ~the above comparison step should be automated~ (descoped for now)
- [x] It is possible for devlopers to compare the generated shoot-spec of KIM and the one generated by the provisioner

**Comment by Disper** [comment]
How shoots specs can be compared using yq:

1. set KUBECONFIG env variable to the gardener's service account's kubeconfig
2.  Download shoot specs you want to compare `kubectl get shoot -n GARDENER_PROJECT_NAME -ojson | jq '.items[] | select(.metadata.name == "SHOOT_NAME")' > SHOOT_NAME.yml`
3. Use `yq` and `diff` to compare them `diff <(yq -P 'sort_keys(..)' -o=props SHOOT_NAME1.yml) <(yq -P 'sort_keys(..)' -o=props SHOOT_NAME2.yml)`

This prints output in following manner:

```
< metadata.name = ag-test
---
> metadata.name = vogfkd3r2s
< spec.kubernetes.enableStaticTokenKubeconfig = true
---
> spec.kubernetes.enableStaticTokenKubeconfig = false
```

**Comment by koala7659** [comment]
PR with feature flag to dump shoot spec created by Provisioner https://github.com/kyma-project/control-plane/pull/3433

**Comment by Disper** [comment]
Blocked by https://github.com/kyma-project/infrastructure-manager/issues/215

**Comment by akgalwas** [comment]
Work plan
- [x] Implement storing shoot files 
  - [ ] KIM
  - [ ] Provisioner
- [ ] Use persistent volume for storing files
  - [ ]  KIM
  - [ ] Provisioner 
- [ ] Enable shoot file storing functionality for DEV environment
  - [ ] KIM
  - [ ] Provisioner
- [ ] Implement shoot comparison tool
  - [ ] Scaffolding (Dockerfile, build etc.)
  - [ ] Implement comparison logic 
    - [ ] Files
    - [ ] Folders
- [ ] Automate comparison
  - [ ] Create job definition that will compare folders containing files created by the KIM, and the Provisioner
